distributed/deploy/tests/test_spec_cluster.py

```diff
@@ -82,6 +83,19 @@ def test_spec_sync(loop):
             assert result == 11
 
 
+def test_no_circular_references(cleanup, loop):
```

Comment:
Why do we need to wait here for the cleanup? Shouldn't the following be sufficient?

```python
import gc
try:
    gc.disable()
    ref = weakref.ref(SpecCluster(scheduler=scheduler, loop=loop))
    assert ref() is None
finally:
    gc.enable()
```

Reply:
Cleanup is generally good practice to include.  It ensures that we don't leak processes or threads between tests.